src/AreaChart/AreaSeries/Area.tsx

```diff
@@ -126,6 +135,22 @@ export const Area: FC<Partial<AreaProps>> = ({
     [interpolation, total]
   );
 
+  const getSegmentAreaPath = useCallback((mousePosition: [x: number, y: number]) => {
+    const [x] = mousePosition;
+
+    const highlightedSegmentCoords = coords.reduce((acc, curr, index) => {
+      if ((curr.x as number) <= x && (coords[index + 1].x as number) >= x) {
+        acc.push(curr, coords[index + 1])
+      }
+
+      return acc
+    }, [])
+
+    const segmentAreaPath = getAreaPath(highlightedSegmentCoords)
+
+    return segmentAreaPath === null ? undefined : segmentAreaPath
+  }, [coords])
```

Comment:
the segment area is being determined using the two coords the mouse lands between, so the highlighted segments are where the graph isn't changing direction.

This isn't exactly what was shown in the issue #74 screenshot, but it was a simpler first stab 👍